src/mem/alloc.h

```diff
@@ -472,13 +480,13 @@ namespace snmalloc
     }
 
     template<Boundary location = Start>
-    static void* external_pointer(void* p)
+    static uintptr_t external_uintptr(void* p)
```

Comment:
I don't like the name of this (what is an external uintptr?).  I think it would be a lot cleaner to define an `address_t` (which is probably `uint32_t` or `uint64_t`) and make this `external_address()`.  We can then insert explicit conversions from addresses to pointers that will run via the pagemap in the CHERI case and be `reinterpret_cast`s in the non-CHERI case.